timeseries/src/autogluon/timeseries/regressor.py

```diff
@@ -13,7 +13,27 @@
 logger = logging.getLogger(__name__)
 
 
-class CovariateRegressor:
+@runtime_checkable
+class CovariateRegressor(Protocol):
+    def is_fit(self) -> bool: ...
+
+    def fit(self, data: TimeSeriesDataFrame, time_limit: Optional[float] = None, **kwargs) -> "CovariateRegressor": ...
+
+    def transform(self, data: TimeSeriesDataFrame) -> TimeSeriesDataFrame: ...
+
+    def fit_transform(
+        self, data: TimeSeriesDataFrame, time_limit: Optional[float] = None, **kwargs
+    ) -> TimeSeriesDataFrame: ...
+
+    def inverse_transform(
+        self,
+        predictions: TimeSeriesDataFrame,
+        known_covariates: TimeSeriesDataFrame,
+        static_features: Optional[pd.DataFrame],
+    ) -> TimeSeriesDataFrame: ...
+
+
+class DefaultCovariateRegressor(CovariateRegressor):
```

Comment:
Nit: Let's rename this (now or in a future PR) into `GlobalCovariateRegressor`? In case we want to add a `LocalCovariateRegressor` in the future that fits a separate regression model for each item.